autoload/asyncomplete/sources/flow.vim

```diff
@@ -66,6 +70,21 @@ function! asyncomplete#sources#flow#get_source_options(opts)
        \ })
 endfunction
 
+let s:flowbin_path_by_dir = {} " dir: <path to flow>
+
+function! s:resolve_flowbin_path(file, fallback)
+    let l:dir = fnamemodify(a:file, ':h')
+    if !has_key(s:flowbin_path_by_dir, l:dir)
+        let l:node_dir = asyncomplete#utils#find_nearest_directory(a:file, 'node_modules')
+        if filereadable(l:node_dir . '/.bin/flow')
```

Comment:
Hm... do you think this can grow so large it can cause a significant memory usage? Intuitively it is just a mapping from strings to strings, note also that the key is a directory, not a filename so the keyspace isn't that huge.

If you still want to cap it — let's go with the higher limit. The operation is very expensive and might just make async nature of the plugin irrelevant if most of the time is spent on traversing the fs to find the flow.